### PR TITLE
LUCENE-10057: Use Lucene abstractions to store KnnVectorDict

### PR DESCRIPTION
**Pull request by msokolov**

### CODE REVIEW

**Comment by uschindler** [comment]
This looks very well, now the dict file is created using IndexOutput and read in using IndexInput. And it is also stored in index directory. I am not sure if storing in index directory is the best idea, because it is not part of segmentinfos, so the index itsself does not know about the file.

But when this gets improved maybe we have a standard FST format available that is "officially part of index".

File: lucene/demo/src/java/org/apache/lucene/demo/IndexFiles.java [hunk 1]
```diff
@@ -153,6 +163,10 @@ public static void main(String[] args) throws Exception {
         // you're done adding documents to it):
         //
         // writer.forceMerge(1);
+      } finally {
+        if (vectorDictInstance != null) {
```
Comment (dweiss) [inline lucene/demo/src/java/org/apache/lucene/demo/IndexFiles.java:167]: IOUtils.close(vararg) can deal with null entries so you can omit the if here.

**Review by dweiss — APPROVED** [review]
Thanks Mike.

File: lucene/demo/src/java/org/apache/lucene/demo/IndexFiles.java (hunk 1 above)
Comment (msokolov) [reply lucene/demo/src/java/org/apache/lucene/demo/IndexFiles.java:167]: ah I didn't realize, thanks I'll change that and push

**Pull request merged**